packages/demo/src/plasma/navigation/SideNavigation.tsx

```diff
@@ -1,71 +1,194 @@
 /* eslint-disable arrow-body-style */
 import React, {useState} from 'react';
-import {RouteChildrenProps, useLocation} from 'react-router';
-import {Link, Route} from 'react-router-dom';
+import {useLocation} from 'react-router';
+import {Link} from 'react-router-dom';
 import {SideNavigation, SideNavigationItem, SideNavigationMenuSection} from 'react-vapor';
 
-import {NavItem, NavItems} from './nav-items';
-
 interface NavLinkProps {
     href?: string;
     label: string;
     disabled?: boolean;
+    isActive?: boolean;
 }
 
-const NavLink: React.FunctionComponent<NavLinkProps> = ({href, label, disabled}) => {
-    return (
-        <Route
-            path={href}
-            children={(routeProps: RouteChildrenProps) => (
-                <SideNavigationItem isActive={!!routeProps.match} disabled={disabled} href={href}>
-                    {disabled ? (
+// Context is more efficient than prop drilling or using useLocation() 65+ times
+const PathnameContext = React.createContext('/');
+
+const NavLink: React.FunctionComponent<NavLinkProps> = ({href, label, disabled, isActive}) => (
+    <PathnameContext.Consumer>
+        {(path) => (
+            <SideNavigationItem disabled={disabled} href={href} isActive={isActive ?? path?.endsWith(href)}>
+                {disabled ? (
+                    <div className="navigation-menu-section">{label}</div>
```

Comment:
Some text should not be put directly in a `div` tag but in a `p` or `li` type tag for example